aspnetcore/mvc/controllers/filters.md

```diff
@@ -14,68 +14,64 @@ uid: mvc/controllers/filters
 
-### Selecting a Filter
+* [Resource filters](#resource-filters) are the first to handle a request after authorization.  They can run code before the rest of the filter pipeline, and after the rest of the pipeline has completed. They're useful to implement caching or otherwise short-circuit the filter pipeline for performance reasons.
 
-[Authorization filters](xref:mvc/controllers/filters#authorization-filters) are used to determine whether the current user is authorized for the request being made.
+* [Action filters](#action-filters) can run code immediately before and after an individual action method is called. They can be used to manipulate the arguments passed into an action and the result returned from the action.
 
-[Resource filters](xref:mvc/controllers/filters#resource-filters) are the first filter to handle a request after authorization, and the last one to touch the request as it is leaving the filter pipeline. They're especially useful to implement caching or otherwise short-circuit the filter pipeline for performance reasons.
+* [Exception filters](#exception-filters) come after Action filters in the pipeline and are used to apply global policies to unhandled exceptions that have occurred up to that point. 
 
-[Action filters](xref:mvc/controllers/filters#action-filters) wrap calls to individual action method calls, and can manipulate the arguments passed into an action as well as the action result returned from it.
+* [Result filters](#result-filters) can run code immediately before and after the execution of individual action results. They run only when the action method has executed successfully and are useful for logic that must surround view or formatter execution.
 
-[Exception filters](xref:mvc/controllers/filters#exception-filters) are used to apply global policies to unhandled exceptions in the MVC app.
+The following diagram shows how these filter types interact in the filter pipeline.
 
-[Result filters](xref:mvc/controllers/filters#result-filters) wrap the execution of individual action results, and only run when the action method has executed successfully. They are ideal for logic that must surround view execution or formatter execution.
+![The request is processed through Authorization Filters, Resource Filters, Model Binding, Action Filters, Action Execution and Action Result Conversion, Exception Filters, Result Filters, and Result Execution. On the way out, the request is only processed by Result Filters and Resource Filters before becoming a response sent to the client.](filters/_static/filter-pipeline-2.png)
 
 ### Implementation
 
-All filters support both synchronous and asynchronous implementations through different interface definitions. Choose the sync or async variant depending on the kind of task you need to perform. They are interchangeable from the framework's perspective.
```

Comment:
@rynowak 

> I'd really prefer to keep something like this in the doc and even make it more prominent. 

I added this sentence to the intro paragraph, but the warning is already there, formatted as a Note following the examples.